### DISCUSSION TITLE
Ошибка SD-card модуля

### DISCUSSION THREAD

**Discussion by MIIRIIK** [vvip-68/LedPanelWiFi#87] [Вопросы и ответы] [answered]
Использую ESP wroom 32, брал по ссылкам, как рекомендовано. Без sd карты все работает, а вот при выставлении в скетче #define USE_SD 1 получаю ошибку, помогите пожалуйста!

In file included from c:\Users\dworkin\OneDrive\���������\Arduino\libraries\SD\src/utility/Sd2Card.h:26,
                 from c:\Users\dworkin\OneDrive\���������\Arduino\libraries\SD\src/utility/SdFat.h:29,
                 from c:\Users\dworkin\OneDrive\���������\Arduino\libraries\SD\src/SD.h:20,
                 from C:\Dod\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:453,
                 from C:\Dod\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:246:
c:\Users\dworkin\OneDrive\���������\Arduino\libraries\SD\src/utility/Sd2PinMap.h:527:2: error: #error Architecture or board not supported.
 #error Architecture or board not supported.
  ^~~~~
grpc: error while marshaling: string field contains invalid UTF-8

Compilation error: grpc: error while marshaling: string field contains invalid UTF-8

**Comment by vvip-68** [comment]
У вас вообще какая-то библиотека  Arduino/libraries/SD - не для ESP платформ, а для ардуино. Удалите ее, она не нужна. Это все в ядре есть и никаких дополнительных библиотек не нужно.

**Reply by MIIRIIK** [reply]
Удалил.
In file included from C:\Dod\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\GyverPanelWiFi_v1.13.ino:246:
C:\Dod\GyverPanelWiFi-master\firmware\GyverPanelWiFi_v1.13\a_def_hard.h:453:10: fatal error: SD.h: No such file or directory
 #include <SD.h>
          ^~~~~~
compilation terminated.
exit status 1

**Reply by vvip-68** [reply, marked as answer]
Значит у вас стоит не та версия ядра, а какая-то древняя, в которой еще нет поддержки SD-карт внутри ядра.

**Comment by MIIRIIK** [comment]
Действительно старое ядро, спасибо!!!! (перешел на другой комп и не обновил)
